network/src/primary.rs

```diff
@@ -72,17 +81,19 @@ impl PrimaryNetwork {
         message: BincodeEncodedPayload,
     ) -> CancelHandler<()> {
         let client = self.client(address);
-        let handle = tokio::spawn(
-            self.retry_config
-                .retry(move || {
-                    let mut client = client.clone();
-                    let message = message.clone();
-                    async move { client.send_message(message).await.map_err(Into::into) }
-                })
-                .map(|response| {
-                    response.expect("we retry forever so this shouldn't fail");
-                }),
-        );
+        let handle = self
+            .spawn(
+                self.retry_config
+                    .retry(move || {
+                        let mut client = client.clone();
```

Comment:
How heavyweight is this `client.clone()`?

Reply:
I don't know, but hopefully it bottoms out in an Arc::clone somwhere, and this precedes my change

Reply:
It's a decorator around a tokio [`Channel`](https://github.com/hyperium/tonic/blob/9ea03c2c44ad6bf76e4141e6fb40870b9b30ab63/tonic/src/transport/channel/mod.rs#L45-L71), TL;DR: not heavyweight.